### PR TITLE
Rule 80 20

### PR DESCRIPTION
**Pull request by Mozand**
#162 80/20 Rule

Lines for the 80/20 rules have been added and by clicking on the horizontal line we can see the corresponding bars.

We have to ask Tibco about the buffer of 80%.

### CODE REVIEW

File: src/stackedBars.ts [hunk 1]
```diff
@@ -48,14 +52,55 @@ export function renderStackedBars(pareto: Pareto, settings: Settings) {
         .attr("shape-rendering", "crispEdges")
         .attr("stroke", (d) => (d.isMarked ? "#000" : "none"))
         .attr("stroke-width", (d) => (d.isMarked ? settings.style.selectionBox.strokeWidth : "0"));
-
+        
+    
+        
+     const range =  svgBoundingClientRect.width - resources.PADDINGRIGHT 
+     const percentageLabelPadding = svgBoundingClientRect.width /4;
+     let ppp = categoryAxisBandwidth.bandwidth()/4;
```
Comment (MajaLinder) [inline src/stackedBars.ts:60]: Can we have a more descriptive variable name?

File: src/stackedBars.ts [hunk 2]
```diff
@@ -48,14 +52,55 @@ export function renderStackedBars(pareto: Pareto, settings: Settings) {
...
+   sel.append("line")
+        .style("stroke", "red")
+        .style("stroke-width", "6")
+        .style("stroke-dasharray", "8 8 ")
+        .attr("x1",  range)
+        .attr("y1", percentageAxis(80))
+        .attr("x2",0)
+        .attr("y2", percentageAxis(80))
+        .on("click",selectedBars)
+        // .on("mouseout", function () {
```
Comment (MajaLinder) [inline src/stackedBars.ts:70]: Why is this outcommented? Is it not used anymore?

File: src/stackedBars.ts [hunk 3]
```diff
@@ -89,6 +134,25 @@ export function renderStackedBars(pareto: Pareto, settings: Settings) {
         markingSelector: ".in-bar"
     });
 
+
+console.log(barCounter)
+    function selectedBars(){
```
Comment (MajaLinder) [inline src/stackedBars.ts:139]: There's an issue with the logic when selecting bars in color by. 

image

**Review by MajaLinder — CHANGES_REQUESTED** [review]
Nice progress! I think however that the red dotted lines were not the actual image that Arnaud was going for. In his answer he said:

 _"I was thinking that if you click on the 80% line (the red line), then the bars that contributed to the 80% should be marked on the chart, like displayed in the example image. **In my image they are here to clarify the concept. I think it should be an additional horizontal line at 80%**"_ 

My understanding is that we should have only one line at the 80% mark, and I think we should clarify the appearance of it.

File: src/stackedBars.ts (hunk 1 above)
Comment (Mozand) [reply src/stackedBars.ts:60]: I changed them

File: src/stackedBars.ts (hunk 2 above)
Comment (Mozand) [reply src/stackedBars.ts:70]: I was waiting for the Tibco comment, now it's fixed.

File: src/resources.ts [hunk 4]
```diff
@@ -7,5 +7,6 @@ export const resources = {
     PADDINGBOTTOMDOWN: 0,
     PADDINGTOPDOWN: 10,
     PADDINGCATEGORYLEFT: 0,
-    PADDINGPERCENTAGEDOWN: 0
+    PADDINGPERCENTAGEDOWN: 0,
+    LINEWIGHT: 3
```
Comment (MajaLinder) [inline src/resources.ts:11]: Great that you added the line weight in the resources file, but there's a typo

**Review by MajaLinder — COMMENTED** [review]
Good work Mo! I just found a small typo.

File: src/resources.ts (hunk 4 above)
Comment (Mozand) [reply src/resources.ts]: just changed!

**Review by MajaLinder — APPROVED** [review]

**Pull request merged**